python/lsst/ts/mtreflector/csc.py

```diff
@@ -0,0 +1,249 @@
+            initial_state=initial_state,
+            override=override,
+            simulation_mode=simulation_mode,
+        )
+
+    @staticmethod
+    def get_config_pkg() -> str:
+        """Get configuration package name."""
+        return "ts_config_mtcalsys"
+
+    @property
+    def connected(self) -> bool:
+        """Is the LabJack connected?"""
+        return self.controller is not None and self.controller.connected
+
+    async def configure(self, config: types.SimpleNamespace) -> None:
+        """Configure the CSC.
+
+        Parameters
+        ----------
+        config : `types.SimpleNameSpace`
+            The config object.
+        """
+        self.config = config
+
+    async def handle_summary_state(self) -> None:
+        """Create connection to labjack when transitioning
+        to disabled state and destroy connection when transitioning
+        to standby state.
+
+        Raises
+        ------
+        ValueError
+            If config is None.
+            If reflector Controller should be connected but isn't.
+        """
+        if self.disabled_or_enabled:
+            if self.should_be_connected and not self.connected:
+                await self.fault(
+                    code=MTReflectorStatus.CONNECTION_ERROR.value,
+                    report="reflector controller should be connected but isn't.",
+                )
+                return
+            if not self.connected:
+
+                if self.config is None:
+                    msg = (
+                        "Tried to create MTReflectorController without a configuration. "
+                        "This is most likely a bug with the control sequence causing "
+                        "the configuration step to be skipped. Try sending the CSC back "
+                        "to STANDBY or OFFLINE and restarting it. You should report this issue."
+                    )
+                    await self.fault(code=1, report=msg)
+                    return
+                self.controller = Controller(
+                    config=self.config,
+                    log=self.log,
+                    simulation_mode=self.simulation_mode,
+                )
+                try:
+                    await self.connect()
+                except Exception:
+                    await self.fault(code=2, report="Failed to connect.")
+                    return
+        else:
+            await self.disconnect()
+
+    async def connect(self) -> None:
+        """Connect to the LabJack and get status.
+
+        This method initiates the MTReflectorController as well.
+
+        Raises
+        ------
+        RuntimeError
+            If there is no MTReflectorController object.
+        asyncio.TimeoutError
+            If it takes longer than self.config.reflector.connect_timeout.
+        """
+        if self.controller is None:
+            raise RuntimeError(
+                "CSC Tried to use reflector controller without a valid object."
```

Comment:
Please improve this error message. Something like `Reflector controller is None.` would be much clearer.